### PR TITLE
Improve coverage

### PR DESCRIPTION
**Pull request by fabwu**

### CODE REVIEW

File: src/Service/UserRegistrationService.php [hunk 1]
```diff
@@ -71,33 +71,14 @@ public function __construct(
     public function onUserRegistration(EventInterface $e)
     {
         $user = $e->getParam('user');
+        $registrationRecord = $this->createRegistrationRecord($user);
```
Comment (ojhaujjwal) [inline src/Service/UserRegistrationService.php]: First of all, changes other than improving code coverages should have been submitted in another PR.
Second, I don't agree with the removal of methods, `sendVerificationEmail` and `sendPasswordRequestEmail`. If the options, `sendVerificationEmail` and `sendPasswordRequestEmail` are false, that means automatic sending of mail is diabled. In that case, these methods are needed to be invoked manually. So, these methods are required.

File: tests/HtUserRegistrationTest/Service/UserRegistrationServiceTest.php [hunk 2]
```diff
@@ -1,29 +1,251 @@
...
     }
-     
+
+    public function testSetPassword()
+    {
+        $password = '1a2b3c';
+        $data['newCredential'] = $password;
+
+        $this->zfcUserEntity->expects($this->once())
+                ->method('setPassword');
```
Comment (ojhaujjwal) [inline tests/HtUserRegistrationTest/Service/UserRegistrationServiceTest.php:230]:
``` php
$this->zfcUserEntity->expects($this->once())
    ->method('setPassword')
    ->with($password);
```

Comment (ojhaujjwal) [inline tests/HtUserRegistrationTest/Service/UserRegistrationServiceTest.php:230]: Oops, my mistake, the password is hashed!

File: src/Service/UserRegistrationService.php (hunk 1 above)
Comment (fabwu) [inline src/Service/UserRegistrationService.php]: Ok this class represent a service :+1: I revert the commit.

**Comment by ojhaujjwal** [comment]
So far, so good.

**Comment by fabwu** [comment]
For the controller tests `bootstrap.php` should adjust like suggested [here](http://framework.zend.com/manual/2.3/en/tutorials/unittesting.html#bootstrapping-your-tests).

@ojhaujjwal Do you agree with that change?

**Comment by ojhaujjwal** [comment]
Keep it as it is. That is not really necessary.

**Pull request merged**